git-tree.py

```diff
@@ -115,7 +120,7 @@ def parse_branches():
         # Build tree contents of the form {"parent": "child"}.
         if branch.name == main_branch_name:
             tree["root"].append(branch.name)
-        elif branch.upstream_branch is None or "origin/" in branch.upstream_branch:
+        elif assume_main_is_upstream(branch.upstream_branch):
             # Assume the branch is downstream of the main branch.
             tree[main_branch_name].append(branch.name)
```

Comment:
It may also be helpful to color the tree prefix differently to indicate that we're making this assumed link.